python/cugraph/sampling/random_walks.py

```diff
@@ -74,29 +70,22 @@ def random_walks(
             start_vertices = G.lookup_internal_vertex_id(start_vertices)
 
     vertex_set, edge_set, sizes = random_walks_wrapper.random_walks(
-        G, start_vertices, max_depth)
+        G, start_vertices, max_depth, use_padding)
 
     if G.renumbered:
         df_ = cudf.DataFrame()
         df_['vertex_set'] = vertex_set
         df_ = G.unrenumber(df_, 'vertex_set', preserve_order=True)
         vertex_set = cudf.Series(df_['vertex_set'])
 
-    edge_list = defaultdict(list)
-    next_path_idx = 0
-    offsets = [0]
-
-    df = cudf.DataFrame()
-    for s in sizes.values_host:
-        for i in range(next_path_idx, s+next_path_idx-1):
-            edge_list['src'].append(vertex_set.values_host[i])
-            edge_list['dst'].append(vertex_set.values_host[i+1])
-        next_path_idx += s
-        df = df.append(edge_list, ignore_index=True)
-        offsets.append(df.index[-1]+1)
-        edge_list['src'].clear()
-        edge_list['dst'].clear()
-    df['weight'] = edge_set
-    offsets = cudf.Series(offsets)
-
-    return df, offsets
+    if use_padding:
+        edge_set_sz = (max_depth-1)*len(start_vertices)
+        return vertex_set, edge_set[:edge_set_sz], sizes
+
+    vertex_set_sz = sizes.sum()
+    edge_set_sz = vertex_set_sz - len(start_vertices)
+    return vertex_set[:vertex_set_sz], edge_set[:edge_set_sz], sizes
+
+
+def rw_path(num_paths, sizes):
```

Comment:
This needs a docstring with example since it's in the user API.

Reply:
updated